src/pokemon_debug.c

```diff
@@ -0,0 +1,754 @@
+                    charDigits[i]--;
+                    break;
+                }
+            }
+        }
+
+        else
+            charDigits[modArrows->currentDigit]--;
+    }
+
+    newValue = CharDigitsToValue(charDigits, modArrows->maxDigits);
+    if (newValue > modArrows->maxValue || newValue < modArrows->minValue)
+    {
+        return FALSE;
+    }
+    else
+    {
+        modArrows->currValue = newValue;
+        for (i = 0; i < MODIFY_DIGITS_MAX; i++)
+             modArrows->charDigits[i] = charDigits[i];
+        return TRUE;
+    }
+}
+
+static void UpdateBattlerValue(struct PokemonDebugMenu *data)
+{
+    u32 i;
+    switch (data->modifyArrows.typeOfVal)
+    {
+    case VAL_U16:
+        *(u16*)(data->modifyArrows.modifiedValPtr) = data->modifyArrows.currValue;
+        break;
+    }
+}
+
+//Sprite functions
+static const struct CompressedSpritePalette *GetMonSpritePalStructCustom(u16 species, bool8 isFemale, bool8 isShiny)
+{
+    if (isShiny)
+    {
+        if (SpeciesHasGenderDifference[species] && isFemale)
+            return &gMonShinyPaletteTableFemale[species];
+        else
+            return &gMonShinyPaletteTable[species];
+    }
+    else
+    {
+        if (SpeciesHasGenderDifference[species] && isFemale)
+            return &gMonPaletteTableFemale[species];
+        else
+            return &gMonPaletteTable[species];
+    }
+}
+
+void BattleLoadOpponentMonSpriteGfxCustom(u16 species, bool8 isFemale, bool8 isShiny, u8 battlerId)
+{
+    u16 paletteOffset;
+    const void *lzPaletteData;
+    const struct CompressedSpritePalette *palette;
+
+    paletteOffset = 0x100 + battlerId * 16;
+
+    palette = GetMonSpritePalStructCustom(species, isFemale, isShiny);
+
+    if (isShiny)
+    {
+        if (SpeciesHasGenderDifference[species] && isFemale)
+            lzPaletteData = gMonShinyPaletteTableFemale[species].data;
+        else
+            lzPaletteData = gMonShinyPaletteTable[species].data;
+    }
+    else
+    {
+        if (SpeciesHasGenderDifference[species] && isFemale)
+            lzPaletteData = gMonPaletteTableFemale[species].data;
+        else
+            lzPaletteData = gMonPaletteTable[species].data;
+    }
+
+    LZDecompressWram(lzPaletteData, gDecompressionBuffer);
+    LoadPalette(gDecompressionBuffer, paletteOffset, 0x20);
+    LoadPalette(gDecompressionBuffer, 0x80 + battlerId * 16, 0x20);
+
+    //if (species == SPECIES_CASTFORM)
+    //{
+    //    paletteOffset = 0x100 + battlerId * 16;
+    //    LZDecompressWram(lzPaletteData, gBattleStruct->castformPalette[0]);
+    //    LoadPalette(gBattleStruct->castformPalette[gBattleMonForms[battlerId]], paletteOffset, 0x20);
+    //}
+
+    // transform's pink color
+    //if (gBattleSpritesDataPtr->battlerData[battlerId].transformSpecies != SPECIES_NONE)
+    //{
+    //    BlendPalette(paletteOffset, 16, 6, RGB_WHITE);
+    //    CpuCopy32(gPlttBufferFaded + paletteOffset, gPlttBufferUnfaded + paletteOffset, 32);
+    //}
```

Comment:
Is this needed? :eyes:

Not a fan of keeping useless code in.

Reply:
No, an oversight - now deleted